sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -493,6 +523,20 @@ public String getId() {
         return Utils.getSimpleObjectMapper().convertValue(object, classType);
     }
 
+    /**
+     * Initializes a new instance of {@link com.azure.cosmos.batch.TransactionalBatch}
+     * that can be used to perform operations across multiple items in the container with the provided partition
+     * key in a transactional manner
+     * <p>
+     * After subscription the operation will be performed.
+     *
+     * @param partitionKey the partition key.
+     * @return A new instance of {@link com.azure.cosmos.batch.TransactionalBatch}.
+     */
+    public TransactionalBatchCore createTransactionalBatch(PartitionKey partitionKey) {
+        return new TransactionalBatchCore(this, partitionKey);
```

Comment:
New public API introduced by this change should be gated behind the @Beta annotation

Reply:
Done. Thanks!